Penteract/Assets/Scripts/Barrel.cpp

```diff
@@ -3,82 +3,110 @@
 #include "Components/Physics/ComponentSphereCollider.h"
 #include "CameraController.h"
 #include "Components/ComponentParticleSystem.h"
+#include "Components/ComponentObstacle.h"
 
 EXPOSE_MEMBERS(Barrel) {
 	// Add members here to expose them to the engine. Example:
-	MEMBER(MemberType::GAME_OBJECT_UID, barrelUID),
-		MEMBER(MemberType::GAME_OBJECT_UID, sphereColliderUID),
-		MEMBER(MemberType::GAME_OBJECT_UID, cameraUID),
-		MEMBER(MemberType::GAME_OBJECT_UID, particlesUID),
-		MEMBER(MemberType::GAME_OBJECT_UID, particlesForTimerUID),
-		MEMBER(MemberType::FLOAT, timeToDestroy),
-		MEMBER(MemberType::FLOAT, timerToDestroy)
-
+	MEMBER(MemberType::FLOAT, timeToDestroy),
+	MEMBER(MemberType::FLOAT, timeWillDoDamage),
+	MEMBER(MemberType::FLOAT, timerToDestroy),
+	MEMBER(MemberType::BOOL, onFloor),
+	MEMBER(MemberType::FLOAT, forceOfFall),
+	MEMBER(MemberType::FLOAT, shakeMultiplier)
 };
 
 GENERATE_BODY_IMPL(Barrel);
 
 void Barrel::Start() {
-	barrel = GameplaySystems::GetGameObject(barrelUID);
+	barrel = &GetOwner();
+	parentTransform = barrel->GetParent()->GetComponent<ComponentTransform>();
+
+	barrelMesh = barrel->GetParent()->GetChild("BarrelMesh");
+	if (barrelMesh) {
+		obstacle = barrelMesh->GetComponent<ComponentObstacle>();
+		if(!onFloor) obstacle->Disable();
+	}
 
-	GameObject* barrelColliderAux = GameplaySystems::GetGameObject(sphereColliderUID);
-	if (barrelColliderAux) {
-		barrelCollider = barrelColliderAux;
+	barrelCollider = barrel->GetParent()->GetChild("Barrel");
+	if (barrelCollider) {
 		barrelCollider->Disable();
 	}
 
-	GameObject* cameraAux = GameplaySystems::GetGameObject(cameraUID);
+	GameObject* cameraAux = GameplaySystems::GetGameObject("Game Camera"); 
```

Comment:
Because we can instantiate these gameobjects and if we refer to an gameobject by UID of a default scene it will not work for other scenes as it does not reference the same gameobject

Reply:
Okay!